src/const_eval.md

```diff
@@ -42,7 +42,9 @@ r[const-eval.const-expr.path-item]
   Recursively defining constants is not allowed.
 
 r[const-eval.const-expr.path-static]
-* Paths to [statics]. These are only allowed within the initializer of a static.
+* Paths to [statics] with these restrictions and observations.
+  * If the evaluation is carried out in an initializer of a `const` item, or inside a `const` block, reads and writes to any `static mut` or [`extern` statics] are not allowed.
```

Comment:
This is missing some things, like array lengths or enum discriminants. I wouldn't try to list them all. Instead, I would say "If the evaluation is *not* carried out in an initializer of a `static` item, ...".

Also, saying `static mut` is wrong since it fails to mention interior mutability. By summary carefully mentioned interior mutability. Please ensure your text matches my [summary](https://github.com/rust-lang/reference/pull/1610#discussion_r1830928182).

```suggestion
  * If the evaluation is *not* carried out in an initializer of a `static` item, then reads and writes to any mutable static (`static mut` or `static` with an interior mutable type) or [`extern` statics] are not allowed.
```

Reply:
Applied with edits. I am also reordering the bullets around.